### PR TITLE
Clef feature

### PR DESCRIPTION
**Pull request by sildater**
added a simple note feature encoding the currently active clef by sign, line, and octave change.

### CODE REVIEW

File: tests/test_clef.py
Comment (manoskary) [inline tests/test_clef.py:22]: I have a couple of questions about this segment:
- Is this test finished? 
- why are you calling the compute_note_array on individual parts instead of the score?
- Would you like to add an assertion at the end validating the values?

File: partitura/musicanalysis/note_features.py
Comment (manoskary) [inline partitura/musicanalysis/note_features.py:573]: By iterating `notes = part.notes_tied` there is no guarantee that the order of the note/features will match the one in `note_array`. Maybe you can add a double check about this where you check if the `n.id == na[i]["id"]` and if not the same then seek the correct note to update the features so it follows the order of the `note_array`.

**Review by manoskary — CHANGES_REQUESTED** [review]
Very nice addition, this would be a very useful feature, there are a couple of comments that need to be addressed before merging this PR.

File: tests/test_clef.py
Comment (sildater) [reply tests/test_clef.py:22]: is not finished

Comment (manoskary) [reply tests/test_clef.py]: Ok, I will wait for you @sildater  to re-request review when ready!

Comment (sildater) [reply tests/test_clef.py]: @manoskary is ready now

**Review by manoskary — CHANGES_REQUESTED** [review]
This PR adresses a Clef feature within note features, it also corrects a number of vulnerabilities at other note_features dependent on the notes_tied instead of the note_array, To ensure correct order a remapping was proposed using a dictionary with note ids, however there is a small comment regarding the uniqueness of note ids.

File: partitura/musicanalysis/note_features.py
Comment (sildater) [reply partitura/musicanalysis/note_features.py:537]: if note or rest IDs are not unique, the value of the last note of that ID will be used in in the output array. That's wrong, but due to the data and non-breaking.

Comment (manoskary) [reply partitura/musicanalysis/note_features.py:537]: In that case, maybe adding a warning would be nice about the feature.

Comment (sildater) [reply partitura/musicanalysis/note_features.py:554]: added two warnings in the top level feature functions

**Review by manoskary — APPROVED** [review]

**Pull request merged**
